### PR TITLE
Fault Tolerance SE Documentation

### PR DESCRIPTION
**Pull request by spericas**
New document describing the FT SE API.

### CODE REVIEW

File: docs/se/faulttolerance/01_faulttolerance.adoc [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+`T` is provided as the argument when invoking this handler. For example:
+
+[source,java]
+----
+Single<Thread> s = Async.create().invoke(() -> Thread.currentThread()));
+s.thenAccept(t -> System.out.println("Async task executed in thread " + t));
+----
+
+The supplier `() -> Thread.currentThread()` is executed in a new thread and
+the value it produces printed by the consumer passed to `thenAccept`.
```
Comment (ljamen) [inline docs/se/faulttolerance/01_faulttolerance.adoc:100]: the value it produces is printed by the consumer and passed to 'thenAccept'.

File: docs/se/faulttolerance/01_faulttolerance.adoc [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    ...
+});
+----
+
+The example above monitors the call to method `mayTakeVeryLong` and reports a
+`TimeoutException` if the execution takes more than 10 milliseconds to complete.
+
+=== Fallbacks
+
+A fall back to a _known_ result can sometimes be an alternative to
```
Comment (ljamen) [inline docs/se/faulttolerance/01_faulttolerance.adoc:166]: A fallback to a _known_ result can ...

fallback is the noun form and fall back is the action

File: docs/se/faulttolerance/01_faulttolerance.adoc [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+configured to monitor such calls and block attempts that are likely to fail, thus improving
+overall performance.
+
+Circuit breakers start in a _closed_ state, letting calls to proceed normally; after
+detecting a certain number of errors during a pre-defined processing window, they can _open_ to
+prevent additional failures. After a circuit has been opened, it can transition
+first to a _half-open_ state before finally transitioning back to a closed state.
+The use of an intermediate state (half-open)
+makes transitions from open to close more progressive, and prevents a circuit breaker
+from eagerly transition states without considering "sufficient" observations.
```
Comment (ljamen) [inline docs/se/faulttolerance/01_faulttolerance.adoc:197]: from eagerly transition states without considering...


from transitioning states without considering

File: docs/se/faulttolerance/01_faulttolerance.adoc [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+Multi<Integer> multi = retry.invokeMulti(() -> Multi.just(0, 1, 2));
+
+IntSubscriber ts = new IntSubscriber();
+multi.subscribe(ts);
+ts.request(Integer.MAX_VALUE);
+----
+
+The call to `Multi.just(0, 1, 2)` simply returns a multi that produces
+the integers 0, 1 and 2. If an error was generated during this process,
+the policy will retry the call one more time &mdash;for a total of 2
```
Comment (ljamen) [inline docs/se/faulttolerance/01_faulttolerance.adoc:340]: does the em dash render correctly?

**Review by ljamen — APPROVED** [review]
This is very well written. I made only a few comments.

File: docs/se/faulttolerance/01_faulttolerance.adoc (hunk 4 above)
Comment (spericas) [reply docs/se/faulttolerance/01_faulttolerance.adoc:340]: Actually, using the view option here on Github, it renders fine.

File: docs/sitegen.yaml [hunk 5]
```diff
@@ -208,6 +208,14 @@ backend:
...
                     - "se/aot/*.adoc"
+            - title: "Fault Tolerance"
+              pathprefix: "/se/faulttolerance"
+              glyph:
+                type: "icon"
+                value: "warning"
+              items:
+                - includes:
+                    - "se/faulttolerance/*.adoc"
         - title: "Helidon MP"
```
Comment (ljamen) [inline docs/sitegen.yaml:219]: Is this meant to be MP here?

**Review by ljamen — APPROVED** [review]
Just a single question re: the MP at the end of the yaml.

File: docs/sitegen.yaml (hunk 5 above)
Comment (spericas) [reply docs/sitegen.yaml:219]: Fault Tolerance SE should be the last entry for SE. The next line is where MP starts as I understand it.

**Review by ljamen — APPROVED** [review]
Looks good to me.

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**